x/commerciokyc/keeper/keeper.go

```diff
@@ -0,0 +1,267 @@
+	types.MembershipTypeGold:   2500,
+	types.MembershipTypeBlack:  50000,
+}
+
+type Keeper struct {
+	Cdc              *codec.Codec
+	StoreKey         sdk.StoreKey
+	SupplyKeeper     supply.Keeper
+	BankKeeper       bank.Keeper
+	governmentKeeper government.Keeper
+	accountKeeper    auth.AccountKeeper
+}
+
+var (
+	// DPY is Days Per Year
+	DPY = sdk.NewDecWithPrec(36525, 2)
+	// HPD is Hours Per Day
+	HPD = sdk.NewDecWithPrec(24, 0)
+	// MPH is Minutes Per Hour
+	MPH = sdk.NewDecWithPrec(60, 0)
+	// BPM Blocks Per Minutes 7 secs x Block
+	BPM = sdk.NewDecWithPrec(9, 0)
+	// BPY is Blocks Per Year
+	BPY = DPY.Mul(HPD).Mul(MPH).Mul(BPM)
+)
+
+// NewKeeper creates new instances of the accreditation module Keeper
+func NewKeeper(cdc *codec.Codec, storeKey sdk.StoreKey, supplyKeeper supply.Keeper, sendKeeper bank.Keeper, governmentKeeper government.Keeper, accountKeeper auth.AccountKeeper) Keeper {
+
+	// ensure commerciomint module account is set
+	if addr := supplyKeeper.GetModuleAddress(types.ModuleName); addr == nil {
+		panic(fmt.Sprintf("%s module account has not been set", types.ModuleName))
+	}
+
+	return Keeper{
+		Cdc:              cdc,
+		StoreKey:         storeKey,
+		SupplyKeeper:     supplyKeeper,
+		BankKeeper:       sendKeeper,
+		governmentKeeper: governmentKeeper,
+		accountKeeper:    accountKeeper,
+	}
+}
+
+// storageForAddr returns a string representing the KVStore storage key for an addr.
+func (k Keeper) storageForAddr(addr sdk.AccAddress) []byte {
+	return append([]byte(types.MembershipsStorageKey), k.Cdc.MustMarshalBinaryBare(addr)...)
+}
+
+// BuyMembership allow to commerciomint and assign a membership of the given membershipType to the specified user.
+// If the user already has a membership assigned, deletes the current one and assigns to it the new one.
+func (k Keeper) BuyMembership(ctx sdk.Context, buyer sdk.AccAddress, membershipType string, tsp sdk.AccAddress, height int64) error {
+	if membershipType == types.MembershipTypeBlack {
+		return sdkErr.Wrap(sdkErr.ErrInvalidAddress, "cannot buy black membership")
+	}
+
+	// Transfer the tsp tokens to government
+	membershipPrice := membershipCosts[membershipType] * 1000000 // Always multiply by one million
+	membershipCost := sdk.NewCoins(sdk.NewInt64Coin("uccc", membershipPrice))
+	govAddr := k.governmentKeeper.GetGovernmentAddress(ctx)
+	if err := k.BankKeeper.SendCoins(ctx, tsp, govAddr, membershipCost); err != nil {
+		return err
+	}
+
+	// Assign the membership
+	return k.AssignMembership(ctx, buyer, membershipType, tsp, height)
+}
+
+// AssignMembership allow to commerciomint and assign a membership of the given membershipType to the specified user.
+// If the user already has a membership assigned, deletes the current one and assigns to it the new one.
+// Returns the URI of the new minted token represented the assigned membership, or an error if something goes w
+// THIS FUNCTION CAN TRANSFORM WITH AssignMembership(ctx sdk.Context, m types.Membership)
```

Comment:
What does this mean?

Reply:
Maybe it is better to pass the whole membership in the function rather than every single element. This is the meaning of the comment